src/test/java/uk/gov/ons/census/casesvc/service/ReceiptProcessorTest.java

```diff
@@ -1,118 +1,98 @@
-    receipt.setResponseDateTime(expectedReceiptDateTime);
-
-    ReceiptProcessor receiptProcessor =
-        new ReceiptProcessor(caseProcessor, uacQidLinkRepository, caseRepository, uacProcessor);
-    receiptProcessor.processReceipt(receipt, headers);
+    underTest.processReceipt(managementEvent);
 
     // then
     verify(uacProcessor, times(1)).emitUacUpdatedEvent(expectedUacQidLink, expectedCase, false);
-    verify(uacProcessor, times(1))
-        .logEvent(
+    verify(eventLogger, times(1))
+        .logReceiptEvent(
             expectedUacQidLink,
             QID_RECEIPTED,
             EventType.UAC_UPDATED,
-            payloadDTO,
-            headers,
-            receipt.getResponseDateTime());
+            expectedReceipt,
+            managementEvent.getEvent(),
+            expectedReceipt.getResponseDateTime());
   }
 
   @Test(expected = RuntimeException.class)
   public void testReceiptedQidNotFound() {
-    // Given
-    CaseRepository caseRepository = mock(CaseRepository.class);
-    CaseProcessor caseProcessor = mock(CaseProcessor.class);
-    UacProcessor uacProcessor = mock(UacProcessor.class);
-    UacQidLinkRepository uacQidLinkRepository = mock(UacQidLinkRepository.class);
-
-    // Given
-    Receipt receipt = new Receipt();
-    receipt.setCaseId(TEST_CASE_ID.toString());
-
-    ReceiptProcessor receiptProcessor =
-        new ReceiptProcessor(caseProcessor, uacQidLinkRepository, caseRepository, uacProcessor);
-    receiptProcessor.processReceipt(receipt, createTestHeaders());
-
-    // Then
-    // Expected Exception is raised
-
-  }
-
-  private UacQidLink getUacQidLink() {
-    UacQidLink uacQidLink = new UacQidLink();
-    uacQidLink.setCaze(null);
-    uacQidLink.setUac(TEST_UAC);
-    uacQidLink.setId(UUID.randomUUID());
-    uacQidLink.setQid(TEST_QID);
-    return uacQidLink;
-  }
-
-  private Case getRandomCase() {
-    EasyRandom easyRandom = new EasyRandom();
-    Case caze = easyRandom.nextObject(Case.class);
-    caze.setCaseId(TEST_CASE_ID);
-
-    return caze;
-  }
-
-  private Map<String, String> createTestHeaders() {
-    Map<String, String> headers = new HashMap<>();
-
-    headers.put("channel", "any receipt channel");
-    headers.put("source", "any receipt source");
-
-    return headers;
+    // GIVEN
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
+    String expectedQuestionnaireId = managementEvent.getPayload().getReceipt().getQuestionnaireId();
+    String expectedErrorMessage =
+        String.format("Questionnaire Id '%s' not found!", expectedQuestionnaireId);
+
+    when(uacQidLinkRepository.findByQid(anyString())).thenReturn(Optional.empty());
+
+    try {
+      // WHEN
+      underTest.processReceipt(managementEvent);
+    } catch (RuntimeException re) {
+      // THEN
+      assertThat(re.getMessage()).isEqualTo(expectedErrorMessage);
+      throw re;
+    }
+
+    fail("Questionnaire Id runtime exception did not throw!");
```

Comment:
fail not needed, if it gets that far it'll fail anyway as it won't get the Exception